fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java

```diff
@@ -237,6 +260,21 @@ public void writeData(long id, int block, @Nonnull byte[] value) {
     @Override
     @Nonnull
     public String[] listAll() {
+        List<String> outList;
+        final FDBStoreTimer timer = context.getTimer();
+        if (timer != null) {
+            long start = System.nanoTime();
+            outList = listAllInternal();
+            timer.record(FDBStoreTimer.Events.LUCENE_LIST_ALL, System.nanoTime() - start);
+        } else {
+            outList = listAllInternal();
+        }
+        //noinspection ToArrayCallWithZeroLengthArrayArgument
+        return outList.toArray(new String[outList.size()]);
```

Comment:
You have a method, above, `incrementCallCount` to centralize the handling of the timer (one of my BIGGEST irritations is that the store timer can be `null` -- we really should have had a no-op implementation of the timer that can stand in for `null`, but ... here we are).  I think it would make sense to similarly have:

```
private void record(StoreTimer.Event event, long durationNanos) {
    final StoreTimer timer = context.getTimer();
    if (timer != null) {
        time.record(event, duration);
    }
}
```

this will greatly simplify this sort of code to just:

```
long start = System.nanoTime()
try {
    return listAllInternal();
} finally {
    record(FDBStoreTimer.Events.LUCENE_LIST_ALL, System.nanoTime() - start);
}
```

I think it's fine to live with the overhead of fetching the time even when a store timer isn't present.